test_case_gen.py

```diff
@@ -0,0 +1,120 @@
+"""
+Generates Test Cases 0, 1 and saves them to json
+"""
+from network_simulator.network_topology import *
+from network_simulator.host import *
+from network_simulator.common import *
+from network_simulator.router import *
+from network_simulator.flow import *
+from network_simulator.link import *
+
+if __name__ == "__main__":
+    # test case 0
+    links = list()
+    l1 = Link("H1", "H2", h1, h2, LinkBuffer(64 * KILOBYTE), 0.01,

```

Comment:
There are a lot of params in these constructors. You should use named variables to refer to each parameter. For example:

```
l1 = Link(end_1_addr="H1", end_2_addr="H2", ...)
```

Also, Links cannot have their end_1_device or end_2_device set. Please read the NetworkTopology preconditions. The JSON files generated by this script should be possible to feed into initializer and work successfully, which is not the case now. Please run this script and check that it works before submitting a new version.